WordPress/src/main/java/org/wordpress/android/ui/JetpackRemoteInstallViewModel.kt

```diff
@@ -0,0 +1,103 @@
+import org.wordpress.android.ui.JetpackRemoteInstallViewState.Installed
+import org.wordpress.android.ui.JetpackRemoteInstallViewState.Start
+import org.wordpress.android.ui.JetpackRemoteInstallViewState.Type
+import org.wordpress.android.ui.JetpackRemoteInstallViewState.Type.ERROR
+import org.wordpress.android.ui.JetpackRemoteInstallViewState.Type.INSTALLED
+import org.wordpress.android.ui.JetpackRemoteInstallViewState.Type.INSTALLING
+import org.wordpress.android.ui.JetpackRemoteInstallViewState.Type.START
+import org.wordpress.android.viewmodel.SingleLiveEvent
+import javax.inject.Inject
+
+class JetpackRemoteInstallViewModel
+@Inject constructor(
+    private val jetpackStore: JetpackStore,
+    private val dispatcher: Dispatcher,
+    private val accountStore: AccountStore,
+    private val siteStore: SiteStore
+) : ViewModel() {
+    private val mutableViewState = MutableLiveData<JetpackRemoteInstallViewState>()
+    val liveViewState: LiveData<JetpackRemoteInstallViewState> = mutableViewState
+    private val mutableJetpackConnectionFlow = SingleLiveEvent<JetpackConnectionData>()
+    val liveJetpackConnectionFlow: LiveData<JetpackConnectionData> = mutableJetpackConnectionFlow
+    private var siteModel: SiteModel? = null
+
+    init {
+        dispatcher.register(this)
+    }
+
+    fun start(site: SiteModel, type: Type?) {
+        siteModel = site
+        // Init state only if it's empty
+        if (mutableViewState.value == null) {
+            mutableViewState.value = type.toState(site)
+        }
+    }
+
+    override fun onCleared() {
+        super.onCleared()
+        dispatcher.unregister(this)
+    }
+
+    fun onLogin(siteId: Int) {
+        connectJetpack(siteId)
+    }
+
+    private fun Type?.toState(site: SiteModel): JetpackRemoteInstallViewState {
+        if (this == null) {
+            return Start { startRemoteInstall(site) }
+        }
+        return when (this) {
+            START -> Start { startRemoteInstall(site) }
+            INSTALLING -> {
+                startRemoteInstall(site)
+                JetpackRemoteInstallViewState.Installing
+            }
+            INSTALLED -> Installed { connectJetpack(site.id) }
+            ERROR -> Error { startRemoteInstall(site) }
+        }
+    }
+
+    private fun startRemoteInstall(site: SiteModel) {
+        mutableViewState.postValue(JetpackRemoteInstallViewState.Installing)
+        dispatcher.dispatch(JetpackActionBuilder.newInstallJetpackAction(site))
+    }
+
+    private fun connectJetpack(siteId: Int) {
+        val site = siteStore.getSiteByLocalId(siteId)
+        val hasAccessToken = accountStore.hasAccessToken()
+        mutableJetpackConnectionFlow.postValue(JetpackConnectionData(site, hasAccessToken))
+    }
+
+    // Network Callbacks
+    @Subscribe(threadMode = ThreadMode.BACKGROUND)
+    @SuppressWarnings("unused")
+    fun onEventsUpdated(event: OnJetpackInstalled) {
+        val site = siteModel ?: return
```

Comment:
tbh I don't think this could ever happen. You'd have to receive an event before the `start` is called and in that case you could be sure it wasn't this VM that triggered the operation (so it's ok to ignore the event). The reason I'm registering in `init` is that you cannot register twice and the start method could be called several times. This way I don't have to worry about registering/deregistering and I also still receive events even when the fragment is stopped (or the activity destroyed) which is exactly what I want.